CodeEdit/Features/NavigatorArea/SourceControlNavigator/SourceControlToolbarBottom.swift

```diff
@@ -23,8 +25,11 @@ struct SourceControlToolbarBottom: View {
 
     private var sourceControlMenu: some View {
         Menu {
-            Button("Discard Changes...") {}
-                .disabled(true) // TODO: Implementation Needed
+            Button("Discard Changes...") {
+                if discardChangesDialog() {
+                    workspace.sourceControlManager?.discardAllChanges()
+                }
+            }
```

Comment:
Is there any possibility this could be `nil` at this point? Im not 100% familiar with state handling on this bit.

Reply:
Ideally no never, but unwrapping this as an optional without error handling is fine.